intel-sgx/ias/src/verifier.rs

```diff
@@ -0,0 +1,280 @@
+            MissingIasReport => f.write_fmt(format_args!("no IAS report extension in certificate")),
+            ReportAsn1Parse => f.write_fmt(format_args!("unable to parse report ASN.1")),
+            ReportJsonParse => f.write_fmt(format_args!("unable to parse report JSON")),
+            ReportBadQuoteSize => f.write_fmt(format_args!("invalid size of enclave quote body")),
+            SpkiHashFailure => f.write_fmt(format_args!("error computing SPKI hash")),
+            ReportNoCertificate => f.write_fmt(format_args!("no signing certificates in report")),
+            ReportInvalidCertificate => f.write_fmt(format_args!("invalid signing certificate in report")),
+            ReportUntrustedCertificate => f.write_fmt(format_args!("server report signing certificate is not trusted")),
+            ReportBadSignature => f.write_fmt(format_args!("bad report signature")),
+            ReportBadQuoteStatus(qstatus, ref pstatus) => f.write_fmt(format_args!("bad quote status in report: {:?}{}", qstatus,
+                match *pstatus {
+                    Some(Ok(ref pstatus)) => format!(" ({})", pstatus),
+                    Some(Err(ref e)) => format!(" (failed to parse platform info blob {})", e),
+                    None => String::new(),
+                }
+            )),
+            ReportBadPayload => f.write_fmt(format_args!("SPKI does not match REPORTDATA")),
+            ReportBadVersion { actual, expected } => f.write_fmt(format_args!("report version is {}, expected {}", actual, expected)),
+            MrsignerMismatch { ref local, ref remote } => f.write_fmt(format_args!("enclave peer mrsigner mismatch, local = {}, remote = {}",
+                                                                                   crate::HexPrint(&local), crate::HexPrint(&remote))),
+            IsvprodidMismatch { local, remote } => f.write_fmt(format_args!("enclave peer isvprodid mismatch, local = {}, remote = {}",
+                                                                            local, remote)),
+            FlagsMismatch { local, remote } =>
+                f.write_fmt(format_args!("enclave peer attributes.flags mismatch, masked local = {:?}, masked remote = {:?}",
+                                         local, remote)),
+            XfrmMismatch { local, remote } =>
+                f.write_fmt(format_args!("enclave peer attributes.xfrm mismatch, masked local = {}, masked remote = {}",
+                                         local, remote)),
+            MiscselectMismatch { local, remote } =>
+                f.write_fmt(format_args!("enclave peer miscselect mismatch, masked local = {:?}, masked remote = {:?}",
+                                         local, remote)),
+        }
+    }
+}
+
+/// Verify that `report` is correctly signed by a key that chains to one of the
+/// trusted certificates in `ca_certificates`.
+///
+/// Does NOT verify the report contents.
```

Comment:
Doc comment should be on the public function

Reply:
Fixed